app/src/main/java/chat/rocket/android/authentication/registerusername/ui/RegisterUsernameFragment.kt

```diff
@@ -6,25 +6,39 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import androidx.core.view.ViewCompat
 import androidx.fragment.app.Fragment
 import chat.rocket.android.R
 import chat.rocket.android.analytics.AnalyticsManager
 import chat.rocket.android.analytics.event.ScreenViewEvent
 import chat.rocket.android.authentication.registerusername.presentation.RegisterUsernamePresenter
 import chat.rocket.android.authentication.registerusername.presentation.RegisterUsernameView
+import chat.rocket.android.util.extension.asObservable
 import chat.rocket.android.util.extensions.inflate
 import chat.rocket.android.util.extensions.setVisible
-import chat.rocket.android.util.extensions.shake
 import chat.rocket.android.util.extensions.showKeyboard
 import chat.rocket.android.util.extensions.showToast
 import chat.rocket.android.util.extensions.textContent
 import chat.rocket.android.util.extensions.ui
-import chat.rocket.android.util.extensions.vibrateSmartPhone
 import dagger.android.support.AndroidSupportInjection
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
 import kotlinx.android.synthetic.main.fragment_authentication_register_username.*
+import java.util.concurrent.TimeUnit
 import javax.inject.Inject
 
-internal const val TAG_REGISTER_USERNAME_FRAGMENT = "RegisterUsernameFragment"
+private const val BUNDLE_USER_ID = "user_id"
+private const val BUNDLE_AUTH_TOKEN = "auth_token"
+
+fun newInstance(userId: String, authToken: String): Fragment {
```

Comment:
Wouldn't be better if we put this method declaration in a `companion object` inside `RegisterUsernameFragment`?

Reply:
I'm not sure. `Companion object` works fine too, but doing that we'll need to make the same for all the others fragments.